aws_lambda_builders/workflows/python_pip/workflow.py

```diff
@@ -73,17 +73,28 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
         if osutils is None:
             osutils = OSUtils()
 
+        self.actions = []
         if osutils.file_exists(manifest_path):
             # If a requirements.txt exists, run pip builder before copy action.
-            self.actions = [
-                PythonPipBuildAction(artifacts_dir, scratch_dir, manifest_path, runtime, binaries=self.binaries),
-                CopySourceAction(source_dir, artifacts_dir, excludes=self.EXCLUDED_FILES),
-            ]
+            if self.download_dependencies:
+                self.actions.append(
+                    PythonPipBuildAction(
+                        artifacts_dir,
+                        scratch_dir,
+                        manifest_path,
+                        runtime,
+                        self.dependencies_dir,
+                        binaries=self.binaries,
+                    )
+                )
+            # if dependencies folder is provided, copy dependencies from dependencies folder to build folder
+            if self.dependencies_dir:
+                self.actions.append(
+                    CopySourceAction(self.dependencies_dir, artifacts_dir, excludes=self.EXCLUDED_FILES)
+                )
```

Comment:
Same comment from the other PR (Ruby, NodeJS). What happens if `self.dependencies_dir` is not defined?